src/pages/Swap.tsx

```diff
@@ -171,14 +181,17 @@ function Swap(): ReactElement {
                   console.log("unknown symbol", { to, swapType })
                   return null
                 }
-                const token = tokenSymbolToTokenMap[to.symbol] as BasicToken
+                if (!tokens) return null
```

Comment:
I think we can set 
`const tokens = setContext(TokensContext) || {}` or something similar so we can get rid of this line as well as the 
`const allTokens = Object.values(tokens || {})` on L138

Reply:
when attempting in this manner it states to wrap the useContext in a useMemo which seems strange.